web/src/message_list_data_cache.ts

```diff
@@ -0,0 +1,33 @@
+import type {Filter} from "./filter";
+import type {MessageListData} from "./message_list_data";
+
+const cache: MessageListData[] = [];
+
+export function get(filter: Filter): MessageListData | undefined {
+    for (const cached_data of cache) {
+        if (cached_data.filter.equals(filter)) {
+            return cached_data;
+        }
+    }
+    return undefined;
+}
+
+export function add(message_list_data: MessageListData): void {
+    for (let i = 0; i < cache.length; i += 1) {
+        // Replace existing data in the cache if the filter is the same.
+        if (cache[i]?.filter.equals(message_list_data.filter)) {
+            cache[i] = message_list_data;
+            return;
+        }
+    }
+
+    cache.push(message_list_data);
+}
+
+export function all(): MessageListData[] {
+    return cache;
+}
+
+export function clear(): void {
+    cache.length = 0;
```

Comment:
Is that a correct way to empty a list object? I would have expected something like `cache = []`

Reply:
Yeah

> The array object observes the length property, and automatically syncs the length value with the array's content. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length

Reply:
Using cache = [all_messages_data.all_messages_data]; now